inductiva/resources/machines.py

```diff
@@ -17,73 +19,67 @@ def __init__(
         spot: bool = False,
         disk_size_gb: int = 20,
         zone: typing.Optional[str] = "europe-west1-b",
+        name: typing.Optional[str] = None,
```

Comment:
Is the idea to use the name to fetch a machine?

After launching a machine, going about your life, returning and wanting to pick up that machine again just by the name, none of the functions `start` or `terminate` will work properly just with `machine_group=resources.MachineGroup(name="api-something")`.

So I am really not understanding the point of setting the name in the argument. You need the `get_machine_group` for this to make sense. Which is exactly what you have in the resources.py.

So it's not clear why should it be passed here.

Reply:
That's exactly why I already added that function [here](https://github.com/inductiva/inductiva/blob/cd30d2b741913cbe58302e1122891ec3ceb8b709/inductiva/resources/resources.py#L44) it retrieves the machine by name and recreates it.

Reply:
Why is the name optional? I was under the impression that the names were created in the backend by us. If it's optional here, can the user choose a name that we will actually use?

Reply:
Yes, but you can use that function and initialise a MachineGroup without setting the name through the init. If you want to pass the name to the client just do it afterwards. I will explain my self better in that part of the code.

Reply:
I removed it now, I found another way to integrate it later when we retrieve it from the backend. So it's not there anymore. And about the labels, yes I think it is very important, we will add them eventually, but I think it's another PR, because it needs to be implemented in the backend as well.